ark/utils/spatial_analysis_utils.py

```diff
@@ -142,12 +171,11 @@ def compute_close_cell_num(dist_mat, dist_lim, num, analysis_type,
     # iterating k from [j, end] cuts out 1/2 the steps (while symmetric)
     for j, m1n in enumerate(mark1_num):
         for k, m2n in enumerate(mark1_num[j:], j):
-            close_num[j, k] = np.sum(
-                dist_mat_bin[np.ix_(
-                    np.asarray(mark1poslabels[j] - 1, dtype='int'),
-                    np.asarray(mark1poslabels[k] - 1, dtype='int')
-                )]
-            )
+            close_num[j, k] = np.sum(dist_mat_bin.loc[
+                np.array(mark1poslabels[j]),
+                np.array(mark1poslabels[k])
+            ].values)
```

Comment:
this is kinda shnice.

one thing tho, do the labels need to be `np.array`s ?

Reply:
No, you should be able to index with lists of ints